lib/galaxy/schema/invocation.py

```diff
@@ -218,3 +250,247 @@ class InvocationMessageResponseModel(BaseModel):
+    )
+    workflow_step_id: EncodedDatabaseIdField = Field(
+        ...,
+        title="Workflow step ID",
+        description="The encoded ID of the workflow step associated with the dataset/dataset collection.",
+    )
+
+
+class InvocationInput(InvocationIOBase):
+    label: Optional[str] = Field(
+        default=Required,
+        title="Label",
+        description="Label of the workflow step associated with the input dataset/dataset collection.",
+    )
+    src: Union[Literal[DataItemSourceType.hda], Literal[DataItemSourceType.hdca]] = Field(
+        default=Required, title="Source", description="Source type of the input dataset/dataset collection."
+    )
+
+
+class InvocationInputParameter(Model):
+    parameter_value: str = Field(default=Required, title="Parameter value", description="Value of the input parameter.")
+    label: str = Field(
+        default=Required, title="Label", description="Label of the workflow step associated with the input parameter."
+    )
+    workflow_step_id: EncodedDatabaseIdField = Field(
+        default=Required,
+        title="Workflow step ID",
+        description="The encoded ID of the workflow step associated with the input parameter.",
+    )
+
+
+class InvocationOutput(InvocationIOBase):
+    src: Literal[DataItemSourceType.hda] = Field(
+        default=Required, title="Source", description="Source model of the output dataset."
+    )
+
+
+class InvocationOutputCollection(InvocationIOBase):
+    src: Literal[DataItemSourceType.hdca] = Field(
+        default=Required, title="Source", description="Source model of the output dataset collection."
+    )
+
+
+class ShowInvocationResponse(BaseModel):
+    id: EncodedDatabaseIdField = Field(
+        default=Required, title="ID", description="The encoded ID of the workflow invocation."
+    )
+    create_time: datetime = CreateTimeField
+    update_time: datetime = UpdateTimeField
+    workflow_id: EncodedDatabaseIdField = Field(
+        title="Workflow ID", description="The encoded Workflow ID associated with the invocation."
+    )
+    history_id: EncodedDatabaseIdField = Field(
+        default=Required,
+        title="History ID",
+        description="The encoded ID of the history associated with the invocation.",
+    )
+    # TODO Using the UUID4 type fails here... why?
+    uuid: Any = Field(..., title="UUID", description="Universal unique identifier of the workflow invocation.")
+    state: InvocationState = Field(
+        default=Required, title="Invocation state", description="State of workflow invocation."
+    )
+    # TODO - I do not know what this does and only copied it. Is this right and what does it do?
+    model_class: schema.INVOCATION_MODEL_CLASS = ModelClassField(schema.INVOCATION_MODEL_CLASS)
+    steps: List[InvocationStep] = Field(
+        default=Required, title="Steps", description="Steps of the workflow invocation."
+    )
```

Comment:
Could you help me on this?

Reply:
> # TODO Using the UUID4 type fails here... why?

What is the failure that you get when using UUID4 from pydantic?

> # TODO - I do not know what this does and only copied it. Is this right and what does it do?

`model_class` just has to have the "literal" name of the class, the ModelClassField is simply a way of avoiding repeating the pydantic.Field information (title, description, etc.) that is similar to all those literals.

Reply:
> What is the failure that you get when using UUID4 from pydantic?
```

pydantic.error_wrappers.ValidationError: 1 validation error for ShowInvocationResponse
uuid
  uuid version 4 expected (type=value_error.uuid.version; required_version=4)
```

Reply:
Can you debug the test that is failing to check the actual value of the `uuid` in the response? could it be None?

Reply:
That seems odd, If it is really a v1 I suggest changing the type then to `Union[UUID4, UUID1]` (or whatever) and add a note about it